### PR TITLE
feat: add `array/base/cunone-by`

### LINKED ISSUE #2325: [RFC]: add `@stdlib/array/base/cunone-by`
### Description

This RFC proposes adding the package `@stdlib/array/base/cunone-by`, which cumulatively tests whether no array element in a provided array passes a test implemented by a predicate function. The function should return a new generic array. The package should also provide an `#assign` API for setting output values in a provided output array.

```javascript
function isPositive( value ) {
	return ( value > 0 );
}

var x = [ 0, 0, 0, 1, 0 ];

var y1 = cunoneBy( x, isPositive );
// returns [ true, true, true, false, false ]

var y2 = [ false, null, false, null, false, null, false, null, false, null ];
var out = cunoneBy.assign( x, y2, 2, 0, isPositive );
// returns [ true, null, true, null, true, null, false, null, false, null ]

var bool = ( out === y2 );
// returns true
```

where the `assign` API supports an offset and stride (see, e.g., `@stdlib/array/base/take`).

Both APIs should support accessor arrays (see, e.g., `@stdlib/array/base/take`).

### Related Issues

No.

### Questions

No.

### Other

- See also `@stdlib/array/base/none-by`

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Naresh-kumar-Thodupunoori**
Resolves #2325 .

## Description

> What is the purpose of this pull request?

This pull request:

-  It adds the package @stdlib/array/base/cunone-by 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #2325 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Naresh-kumar-Thodupunoori** [comment]
@kgryte can you please review this !!!!!

File: lib/node_modules/@stdlib/array/base/cunone-by/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/README.md:1]: This README needs to be updated, as it does not match the style and conventions of other similar files in this project. Please update.

File: lib/node_modules/@stdlib/array/base/cunone-by/benchmark/benchmark.assign.length.js
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/benchmark/benchmark.assign.length.js:1]: The indentation of this file is off.

File: lib/node_modules/@stdlib/array/base/cunone-by/benchmark/benchmark.assign.length.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	max = 6; // 10^max
+
+	for ( i = min; i <= max; i++ ) {
+		len = pow( 10, i );
+		f = createBenchmark( len );
+		bench( pkg+':assign:len='+len, f );
+	}
+}
+
+main();
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/benchmark/benchmark.assign.length.js:98]: You need to add trailing newlines. Please ensure EditorConfig is installed as discussed in the contributing guidelines.

File: lib/node_modules/@stdlib/array/base/cunone-by/benchmark/benchmark.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        y = cunoneBy.assign( x, out, 1, 0, predicate );
+        if ( typeof y !== 'object' ) {
+            b.fail( 'should return an array' );
+        }
+    }
+    b.toc();
+    if ( !isArray( y ) || y !== out ) {
+        b.fail( 'should return the output array' );
+    }
+    for ( i = 0; i < y.length; i++ ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/benchmark/benchmark.js:81]: This check is unnecessary. Remove.

File: lib/node_modules/@stdlib/array/base/cunone-by/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{{alias}}( x, predicate )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/docs/repl.txt:1]: Match the conventions used elsewhere in this project.

File: lib/node_modules/@stdlib/array/base/cunone-by/docs/types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    * @returns output array
+    *
+    * @example
+    * var x = [ 1, 2, 3, 4, 5 ];
+    * var y = [ false, null, false, null, false, null, false, null, false, null ];
+    *
+    * var arr = cunoneBy.assign( x, y, function isEven( v ) { return v % 2 === 0; }, 2, 0 );
+    * // returns [ true, null, false, null, false, null, false, null, false, null ]
+    */
+    assign<T, U extends Collection<boolean> | AccessorArrayLike<boolean>>(
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/docs/types/index.d.ts:69]: Don't line break. The predicate should be the last argument.

File: lib/node_modules/@stdlib/array/base/cunone-by/docs/types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+* var x = [ 1, 2, 3, 4, 5 ];
+*
+* var result = cunoneBy( x, function isEven( v ) { return v % 2 === 0; } );
+* // returns [ true, false, false, false, false ]
+*
+* @example
+* var x = [ 1, 2, 3, 4, 5 ];
+* var y = [ false, null, false, null, false, null, false, null, false, null ];
+*
+* var arr = cunoneBy.assign( x, y, function isEven( v ) { return v % 2 === 0; }, 2, 0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/docs/types/index.d.ts:95]: No inline function declarations. Move out of the invocation.

File: lib/node_modules/@stdlib/array/base/cunone-by/lib/assign.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/lib/assign.js:2]: Missing license.

File: lib/node_modules/@stdlib/array/base/cunone-by/lib/assign.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    if ( isAccessorArray( x ) ) {
+        getter = accessorGetter( x );
+    }
+
+    ix = offset;
+    prevResult = true;
+
+    for ( i = 0; i < x.length; i++ ) {
+        if ( i === 0 ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/lib/assign.js:45]: Why do we need to special case the first argument?

File: lib/node_modules/@stdlib/array/base/cunone-by/lib/index.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+'use strict';
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/lib/index.js:1]: License headers here and everywhere.

File: lib/node_modules/@stdlib/array/base/cunone-by/lib/main.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+function cunoneBy( x, predicate ) {
+    var getter;
+    var out;
+    var i;
+
+    if ( isAccessorArray( x ) ) {
+        getter = accessorGetter( x );
+    }
+
+    out = new Array( x.length );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/lib/main.js:36]: Do not preallocate arrays like this. You can cause the array to be allocated in object mode. Instead use [] and push.

File: lib/node_modules/@stdlib/array/base/cunone-by/package.json [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    "every",
+    "all",
+    "validate"
+  ],
+  "__stdlib__": {},
+  "funding": {
+    "type": "opencollective",
+    "url": "https://opencollective.com/stdlib"
+  }
+}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/package.json:89]: Trailing newlines here and everywhere.

File: lib/node_modules/@stdlib/array/base/cunone-by/test/test.assign.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    t.deepEqual( actual, expected, 'returns expected value' );
+    t.end();
+
+    function isPositive( value ) {
+        return value > 0;
+    }
+});
+
+tape( 'the function handles arrays with all elements failing the test', function test( t ) {
+    var x = [ -1, -2, -3, -4, -5 ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/test/test.assign.js:107]: The order of declarations violates our lint rules. Make sure you have run make init to ensure automation runs on commit and push.

File: lib/node_modules/@stdlib/array/base/cunone-by/test/test.assign.js [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    var actual = cunoneBy( x, isPositive );
+    t.deepEqual( actual, expected, 'returns expected value' );
+    t.end();
+
+    function isPositive( value ) {
+        return value > 0;
+    }
+});
+
+// Add more tests as needed
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/test/test.assign.js:130]: Why is this comment here?

File: lib/node_modules/@stdlib/array/base/cunone-by/test/test.js [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+    function badValue( value ) {
+        return function badValue() {
+            cunoneBy( [1, 2, 3], value );
+        };
+    }
+});
+
+tape( 'the function returns an array', function test( t ) {
+    var out = cunoneBy( [1, 2, 3], function() { return true; } );
+    t.strictEqual( Array.isArray(out), true, 'returns an array' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/test/test.js:77]: We don't use the builtin isArray. Use our utility package.

File: lib/node_modules/@stdlib/array/base/cunone-by/test/test.js [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+    t.end();
+
+    function badValue( value ) {
+        return function badValue() {
+            cunoneBy.assign( value, [], 1, 0, function() {} );
+        };
+    }
+});
+
+// Add more tests as needed for the assign method and other aspects of the main export
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/cunone-by/test/test.js:108]: Either add tests or remove.

**Review by kgryte — CHANGES_REQUESTED** [review]
Left an initial review. Main comment is that you need to carefully follow project conventions. Please try to ensure consistency with the rest of the project.

**Comment by kgryte** [comment]
Additionally all build failures need to be addressed before this PR can be merged or considered for further review.

File: lib/node_modules/@stdlib/array/base/cunone-by/README.md
Comment (Naresh-kumar-Thodupunoori) [reply lib/node_modules/@stdlib/array/base/cunone-by/README.md:1]: I am working on it, actually having some examinations up my sleeve.So preparing for them.

**Comment by Planeshifter** [comment]
Closing as another PR with this package has meanwhile been merged. Thanks for your efforts and interest in `stdlib`  either way!

**Pull request closed without merging**